enhancements/single-node-production-deployment-approach.md

```diff
@@ -0,0 +1,484 @@
+support changing the replica count or disabling the guard entirely,
+depending on the best approach to honor the high-availability mode
+setting in the capabilities API.
+
+#### cluster-ingress-operator
+
+By default, the `cluster-ingress-operator` deploys the router with 2
+replicas. On a single node one will fail to start and the ingress will
+show as degraded. The operator will need to be updated to honor the
+new capabilities API to change the replica count to 1 when the
+high-availability mode is none.
+
+#### cluster-machine-approver
+
+The operator should disable automatic machine approval. (See risks
+section below.)
+
+#### operator-lifecycle-manager
+
+1. OLM operators may be run on other Kubernetes distributions, so they
+   should not rely on an OpenShift API. We need a way to pass the HA
+   mode flag to those operators. One possible solution is using an enviroment
+   variable but needs to discussed further.
+
+### Risks and Mitigations
+
+*How will security be reviewed and by whom? How will UX be reviewed and by whom?*
+
+#### Control plane rollouts
+
+kube-apiserver is periodically reconfigured by its operator in
+response to changes in cluster state (e.g. weekly rotation of etcd
+encryption keys). When this occurs, the apiserver may be inaccessible
+for up to 2 minutes. Workloads that do not depend on the apiserver
+would be expected to run without issue during this interval. Workloads
+that do depend on apiserver availability would need to be resilient to
+these events. OpenShift core components are already resilient in this
+way.
+
+#### Cluster Machine Approver
+
+Auto-approval of certificate signing requests requires 2 sources of
+truth to avoid security attacks like
+[kubeletmein](https://github.com/openshift/machine-config-operator/issues/731). In
+single-node deployments we do not have a second source of truth, and
+need to disable the machine-approver-operator. An outside tool can be
+used to approve any certificate signing requests instead.
+
+#### Lack of high-availability
+
+In a single-node configuration OpenShift would have limited ability to
+provide resiliency features. We will need to clearly communicate these
+limitations to customers choosing to use single-node deployments.
+
+#### Failure to reboot
+
+It is possible that a single node may fail to reboot due to an issue
+with configuration, hardware, or environment. These single-node
+deployments will be managed by external tools, and as with other
+failures related to providing resilient operation, host boot failures
+will need to be detected and mitigated by those tools, rather than the
+local OpenShift instance.
+
+## Design Details
+
+### Open Questions
+
+1. Telco workloads typically require special network setups
+   for a host to boot, including bonded interfaces, access to multiple
+   VLANs, and static IPs. How do we anticipate configuring those?
+2. How do we hand off ownership of the `etcd-quorum-guard` Deployment
```

Comment:
@mrunalp @staebler I've added an open question about the CVO-CEO hand-off that we discussed earlier. Did we have an answer for that, or is it still an open question?

Reply:
@romfreiman mentioned that he and @hexfusion have discussed it and have a plan.

Reply:
is there a link?

Reply:
@romfreiman @hexfusion can you summarize the plan in a comment or provide a link to a design doc so I can put either/both into this enhancement?